modules/dop/initialize.go

```diff
@@ -592,3 +596,70 @@ func copyTestFileTask(ep *endpoints.Endpoints) {
 	}
 	ep.TestSetService().CopyTestSet(record)
 }
+
+// updateCmsNsConfigs update pipeline cms according to pipeline cron which enable is true
+// it will be deprecated in the later version
+func updateCmsNsConfigs(ep *endpoints.Endpoints) error {
+	enable := true
+	// get total
+	cron, err := bdl.Bdl.PageListPipelineCrons(apistructs.PipelineCronPagingRequest{
+		AllSources: true,
+		Sources:    nil,
+		YmlNames:   nil,
+		PageSize:   1,
+		PageNo:     1,
+		Enable:     &enable,
+	})
+	if err != nil {
+		logrus.Errorf("failed to PageListPipelineCrons, err: %s", err.Error())
+		return err
+	}
+	total := cron.Total
+	pageSize := 1000
+	crons := make([]*apistructs.PipelineCronDTO, 0, total)
+	for i := 0; i < int(total)/pageSize+1; i++ {
+		cron, err = bdl.Bdl.PageListPipelineCrons(apistructs.PipelineCronPagingRequest{
+			AllSources: true,
+			Sources:    nil,
+			YmlNames:   nil,
+			PageSize:   pageSize,
+			PageNo:     i + 1,
+			Enable:     &enable,
+		})
+		if err != nil {
+			logrus.Errorf("failed to PageListPipelineCrons, err: %s", err.Error())
+			return err
+		}
+		crons = append(crons, cron.Data...)
+	}
+
+	// userOrgMap judge the user ns is updated or not in the org
+	// key: userID-orgID, value: struct{}
+	userOrgMap := make(map[string]struct{})
```

Comment:
How to judge already compensated?

Reply:
There may be the same org and user in the pipeline cron, and ensuring that there is only one to compensate

Reply:
If a cron record already been compensated, next time dop restart will trigger update again and update time_updated field unexpected. How to avoid update cron record already been compensated?

And add a screenshot about total update time cost and db load.

Reply:
now add etcd lock to ensure that it is executed only once